File: app/src/main/java/com/ahmed/sampleApp/ViewPagerAdapter.java
package com.ahmed.sampleApp;

import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentPagerAdapter;

import java.util.ArrayList;

public class ViewPagerAdapter extends FragmentPagerAdapter {


    public final ArrayList<Fragment> fragments = new ArrayList<>();
    private final ArrayList<String> titles = new ArrayList<>();

    public ViewPagerAdapter(FragmentManager fm) {
        super(fm);
    }

    public void addFragment(Fragment fragment,String title){
        fragments.add(fragment);
        titles.add(title);
    }
    public void addFragment(Fragment fragment){
        fragments.add(fragment);
    }
    public ArrayList<String> getTitles(){
        return titles;
    }


    @Override
    public CharSequence getPageTitle(int position) {
        if (titles.size() > 0){
            return titles.get(position);
        }else {
            return null;
        }
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }
}

File: library/src/main/java/com/ahmed/library/ViewPagerAdapter.kt
package com.ahmed.library

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import java.util.*

class ViewPagerAdapter(fm: FragmentManager, behavior: Int) : FragmentPagerAdapter(fm, behavior) {

    var fragments = ArrayList<Fragment?>()

    fun addAllFragments(fragments: ArrayList<Fragment?>) {
        this.fragments = fragments
        notifyDataSetChanged()
    }

    override fun getItem(position: Int): Fragment {
        return fragments[position]!!
    }

    override fun getCount(): Int {
        return fragments.size
    }
}